test/redteam/commands/generate.test.ts

```diff
@@ -65,6 +65,7 @@ jest.mock('../../../src/providers', () => ({
       cleanup: jest.fn(),
     },
   ]),
+  getProviderIds: jest.fn().mockReturnValue(['test-provider']),
```

Comment:
can you avoid this mock and use the actual implementation? Seems like there isn't anything in that function that would need to be mocked

Reply:
Good q. It needs to be mocked because `loadApiProviders` is mocked. Otherwise execution will enter a state where provider is defined but the list of providers defined in the config is undefined, resulting in `getProviderIds` throwing an error.  In practice this state is never reached because checking for provider ids relies on the existence of a provider.